packages/server/test/plugins/graphql.ts

```diff
@@ -22,24 +23,52 @@ const graphqlChaiPlugin: Chai.ChaiPlugin = (_chai, utils) => {
       const { negate, object } = this.__flags
       const { errors } = object
 
+      const shouldNotHaveErrors = negate
+      const shouldHaveErrors = !negate
+
       const errorsArr = errors || []
 
-      if (negate) {
-        new Assertion(errorsArr).to.have.lengthOf(0)
-      } else {
-        new Assertion(errorsArr).to.have.lengthOf.greaterThanOrEqual(1)
-      }
+      try {
+        if (shouldNotHaveErrors) {
+          new Assertion(errorsArr).to.have.lengthOf(0)
+        } else {
+          new Assertion(errorsArr).to.have.lengthOf.greaterThanOrEqual(1)
+        }
 
-      if (matchMessage) {
-        if (negate) {
-          new Assertion(
-            errorsArr.map((e) => e.message.toLowerCase()).join('\n')
-          ).to.not.contain(matchMessage.toLowerCase())
+        if (matchMessage) {
+          if (shouldNotHaveErrors) {
+            new Assertion(
+              errorsArr.map((e) => e.message.toLowerCase()).join('\n')
+            ).to.not.contain(matchMessage.toLowerCase())
+          } else {
+            new Assertion(
+              errorsArr.map((e) => e.message.toLowerCase()).join('\n')
+            ).to.contain(matchMessage.toLowerCase())
+          }
+        }
+      } catch (e) {
+        if (!(e instanceof AssertionError)) {
+          throw e
+        }
+
+        let msg = ''
+        if (shouldHaveErrors) {
+          if (matchMessage) {
+            msg = `Expected GraphQL response to have errors containing "${matchMessage}", but found none`
+          } else {
+            msg = 'Expected GraphQL response to have errors, but found none'
+          }
         } else {
-          new Assertion(
-            errorsArr.map((e) => e.message.toLowerCase()).join('\n')
-          ).to.contain(matchMessage.toLowerCase())
+          if (matchMessage) {
+            msg = `Expected GraphQL response to have no errors containing "${matchMessage}", but found some`
+          } else {
+            msg = 'Expected GraphQL response to have no errors, but found some'
+          }
+
+          msg += `\nErrors: ${JSON.stringify(errorsArr, null, 2)}`
         }
+
+        throw new AssertionError(msg)
```

Comment:
The current error messages coming out from this assertion suck and don't actually tell you what the errors are